samples/kvssink_intermittent_sample.cpp

```diff
@@ -0,0 +1,293 @@
+        GstEvent* flush_stop = gst_event_new_flush_stop(true);
+        gst_element_send_event(pipeline, flush_stop);
+    }
+    LOG_DEBUG("[KVS sample] Exited stopStartLoop");
+}
+
+int main (int argc, char *argv[])
+{
+    signal(SIGINT, sigint_handler);
+
+    CustomData customData;
+    GstElement *pipeline, *source, *clock_overlay, *video_convert, *source_filter, *encoder, *sink_filter, *kvssink;
+    GstCaps *source_caps, *sink_caps;
+    GstBus *bus;
+    guint bus_watch_id;
+    StreamSource source_type;
+    char stream_name[MAX_STREAM_NAME_LEN + 1];
+
+    gst_init (&argc, &argv);
+
+
+    /* Parse input arguments */
+
+    // Check for invalid argument count.
+    if (argc > 3 || argc < 2) {
+        LOG_ERROR("[KVS sample] Invalid argument count");
+        LOG_INFO("[KVS sample] Usage: " << argv[0] << " <streamName (required)> <testsrc or devicesrc>");
+        return -1;
+    }
+
+    // Get stream name.
+    STRNCPY(stream_name, argv[1], MAX_STREAM_NAME_LEN);
+    stream_name[MAX_STREAM_NAME_LEN] = '\0';
+
+    // Get source type.
+    if(argc > 2) {
+        if(0 == STRCMPI(argv[2], "testsrc")) {
+            LOG_INFO("[KVS sample] Using test source (videotestsrc)");
+            source_type = TEST_SOURCE;
+        } else if (0 == STRCMPI(argv[2], "devicesrc")) {
+            LOG_INFO("[KVS sample] Using device source (autovideosrc)");
+            source_type = DEVICE_SOURCE;
+        } else {
+            LOG_ERROR("[KVS sample] Invalid source type");
+            LOG_INFO("[KVS sample] Usage: " << argv[0] << " <streamName (required)> <testsrc or devicesrc>");
+            return -1;
+        }
+    } else {
+        LOG_ERROR("[KVS sample] No source specified, defualting to test source (videotestsrc)");
+        source_type = TEST_SOURCE;
+    }
+
+
+    /* Create GStreamer elements */
+
+    pipeline = gst_pipeline_new ("kvs-pipeline");
+
+    /* source */
+    if(source_type == TEST_SOURCE) {
+        source = gst_element_factory_make ("videotestsrc", "test-source");
+        g_object_set(G_OBJECT(source), "is-live", TRUE, NULL);
+    } else if (source_type == DEVICE_SOURCE) {
+        source = gst_element_factory_make ("autovideosrc", "device-source");
+    } 
+
+    /* clock overlay */
+    clock_overlay = gst_element_factory_make("clockoverlay", "clock_overlay");
+
+    /* video convert */
+    video_convert = gst_element_factory_make("videoconvert", "video_convert");
+
+    /* source filter */
+    source_filter = gst_element_factory_make("capsfilter", "source-filter");
+    source_caps = gst_caps_new_simple("video/x-raw", "format", G_TYPE_STRING, "I420", NULL);
+    g_object_set(G_OBJECT(source_filter), "caps", source_caps, NULL);
+    gst_caps_unref(source_caps);
+
+    /* encoder */
+    encoder = gst_element_factory_make("x264enc", "encoder");
+    g_object_set(G_OBJECT(encoder), "bframes", 0, NULL);
```

Comment:
We should also set key int max to make sure too long fragments don't get generated